packages/react-redux/modules/components/configure/configure.tsx

```diff
@@ -1,76 +1,50 @@
 import React from 'react';
-import { connect } from 'react-redux';
-import memoize from 'lodash/memoize';
 import { ConfigureAdapter } from '@flopflip/react';
 import {
   Flags,
   Adapter,
-  AdapterArgs,
-  AdapterEventHandlers,
-  AdapterStatus,
+  ConfigureAdapterProps,
   ConfigureAdapterChildren,
 } from '@flopflip/types';
-import { UpdateFlagsAction, UpdateStatusAction } from '../../types';
-import { updateStatus, updateFlags } from './../../ducks';
+import { useUpdateFlags, useUpdateStatus } from '../../hooks';
 
-type Props = {
+type BaseProps = {
   children?: ConfigureAdapterChildren;
   shouldDeferAdapterConfiguration?: boolean;
   defaultFlags?: Flags;
-  adapterArgs: AdapterArgs;
-  adapter: Adapter;
 };
-type ConnectedProps = {
-  handleUpdateStatus: (status: AdapterStatus) => UpdateStatusAction;
-  handleUpdateFlags: (flags: Flags) => UpdateFlagsAction;
-};
-type State = {
-  flags: Flags;
-};
-
-const createAdapterArgs = memoize(
-  (adapterArgs: AdapterArgs, eventHandlers: AdapterEventHandlers) => ({
-    ...adapterArgs,
-    ...eventHandlers,
-  })
-);
+type Props<AdapterInstance extends Adapter> = BaseProps &
+  ConfigureAdapterProps<AdapterInstance>;
 
-export class Configure extends React.PureComponent<
-  Props & ConnectedProps,
-  State
-> {
-  static displayName = 'ConfigureFlopflip';
-
-  static defaultProps = {
-    defaultFlags: {},
-    shouldDeferAdapterConfiguration: false,
-  };
+const defaultProps: Pick<
+  BaseProps,
+  'defaultFlags' | 'shouldDeferAdapterConfiguration'
+> = {
+  defaultFlags: {},
+  shouldDeferAdapterConfiguration: false,
+};
 
-  render(): React.ReactNode {
-    return (
-      <ConfigureAdapter
-        adapter={this.props.adapter}
-        adapterArgs={createAdapterArgs(this.props.adapterArgs, {
-          onFlagsStateChange: this.props.handleUpdateFlags,
-          onStatusStateChange: this.props.handleUpdateStatus,
-        })}
-        defaultFlags={this.props.defaultFlags}
-        shouldDeferAdapterConfiguration={
-          this.props.shouldDeferAdapterConfiguration
-        }
-      >
-        {this.props.children}
-      </ConfigureAdapter>
-    );
-  }
-}
+const Configure = <AdapterInstance extends Adapter>(
+  props: Props<AdapterInstance>
+) => {
+  const handleUpdateFlags = useUpdateFlags();
+  const handleUpdateStatus = useUpdateStatus();
```

Comment:
This will be breaking to flopflip as we require only 16.3 as a peer dependency of react so far. Hooks are 16.8. It's totally fine by me and about time in a way. However, wanted to note :)

Reply:
I see. If you want, we can still be backwards compatible.

If you're ok with breaking changes, we should probably do some more "cleanup":
- remove depreacted lifecycle methods
- remove APIs that might not be necessary anymore
- rewrite components to functional components
- use hooks, rewrite HOCs using hooks
- ??

We can also work on a different base branch.

So if you want to proceed in this direction, I can refactor this PR to avoid using hooks, we can then merge this and start working on the breaking changes on a different base branch.

Reply:
To me this PR as it stands is totally worth it. We give and take a bit. Give improved typings on adapters and take away from React/redux support. Things you've mentioned above are also on my list but all relate on the same dropping of underlying React support to my understanding. 

Having said that I think we can break here once. I actually like breaking releases without much else. Just a breaking change and maybe a small improvement for bait 😄. 

In follow ups we can split out:

1. HoCs to build on hooks
2. Remove deprecated lifecyle (componentWillReceiveProps)
   - This can get a bit hairy as it's on react/configure which is a bit the brain of the library
3. Rewrite anything we can to functional components

Also almost in that order as one enables the other. I just haven't got to it as it's mostly maintainance work and it's odd when your cute little puppy Open Source project transitions into a 80% maintainance hole 🐼. I'd obviously appriciate any help.

As a last step we can then have a look if some APIs should just go away. Candidates for me are `render` in favor of `children` just like `formik` with a deprecation. 

What do you think?

Reply:
I started a board and created issues: https://github.com/tdeekens/flopflip/projects/1